cli/cmd/connect.go

```diff
@@ -155,46 +205,75 @@ func parseCredentialsURI(str string) (string, string, string) {
 	return newStr, credentialsSlice[0], credentialsSlice[1]
 }
 
+// makeConnOpts makes and returns connect options from the arguments.
+func makeConnOpts(network, address string, connCtx connect.ConnectCtx) connector.ConnectOpts {
+	ssl := connector.SslOpts{
+		KeyFile:  connCtx.SslKeyFile,
+		CertFile: connCtx.SslCertFile,
+		CaFile:   connCtx.SslCaFile,
+		Ciphers:  connCtx.SslCiphers,
+	}
+	return connector.ConnectOpts{
+		Network:  network,
+		Address:  address,
+		Username: connCtx.Username,
+		Password: connCtx.Password,
+		Ssl:      ssl,
+	}
+}
+
 // resolveConnectOpts tries to resolve the first passed argument as an instance
 // name to replace it with a control socket or as a URI with/without
 // credentials.
+// It returns connection options and the remaining args.
 func resolveConnectOpts(cmdCtx *cmdcontext.CmdCtx, cliOpts *config.CliOpts,
-	connectCtx *connect.ConnectCtx, args []string) ([]string, error) {
-	newArgs := args
+	connectCtx connect.ConnectCtx, args []string) (
+	connOpts connector.ConnectOpts, newArgs []string, err error) {
 
+	newArgs = args[1:]
 	// FillCtx returns error if no instances found.
 	var runningCtx running.RunningCtx
 	if fillErr := running.FillCtx(cliOpts, cmdCtx, &runningCtx, args); fillErr == nil {
 		if len(runningCtx.Instances) > 1 {
-			return newArgs, fmt.Errorf("specify instance name")
+			err = fmt.Errorf("specify instance name")
+			return
 		}
 		if connectCtx.Username != "" || connectCtx.Password != "" {
-			return newArgs, fmt.Errorf("username and password are not supported" +
+			err = fmt.Errorf("username and password are not supported" +
 				" with a connection via a control socket")
+			return
 		}
-		newArgs[0] = runningCtx.Instances[0].ConsoleSocket
-		return newArgs, nil
-	} else if isCredentialsURI(newArgs[0]) {
+		connOpts = makeConnOpts(
+			connector.UnixNetwork, runningCtx.Instances[0].ConsoleSocket, connectCtx)
+		return
+	} else if isCredentialsURI(args[0]) {
 		if connectCtx.Username != "" || connectCtx.Password != "" {
-			return newArgs, fmt.Errorf("username and password are specified with" +
+			err = fmt.Errorf("username and password are specified with" +
 				" flags and a URI")
+			return
 		}
-		uri, user, pass := parseCredentialsURI(newArgs[0])
-		newArgs[0] = uri
+		newURI, user, pass := parseCredentialsURI(args[0])
+		network, address := parseBaseURI(newURI)
 		connectCtx.Username = user
 		connectCtx.Password = pass
-		return newArgs, nil
-	} else if isBaseURI(newArgs[0]) {
+		connOpts = makeConnOpts(
+			network, address, connectCtx)
```

Comment:
Please, be consistent with existing code style in `tt` codebase: break line before 100-th character. If I see `(` at the end of the line, I expect to see closing `)` on a separate line also.